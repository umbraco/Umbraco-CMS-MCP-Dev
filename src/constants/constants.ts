export const BLANK_UUID = "00000000-0000-0000-0000-000000000000";
export const ROOT_DOCUMENT_TYPE_ID = "a95360e8-ff04-40b1-8f46-7aa4b5983096";
export const CONTENT_DOCUMENT_TYPE_ID = "b871f83c-2395-4894-be0f-5422c1a71e48";
export const Default_Memeber_TYPE_ID = "d59be02f-1df9-4228-aa1e-01917d806cda";
export const TextString_DATA_TYPE_ID = "0cc0eba1-9960-42c9-bf9b-60e150b429ae";
export const IMAGE_MEDIA_TYPE_ID = "cc07b313-0843-4aa8-bbda-871c8da728c8";
export const FOLDER_MEDIA_TYPE_ID = "f38bd2d7-65d0-48e6-95dc-87ce06ec2d3d";
export const EXAMPLE_IMAGE_PATH =
  "/src/umb-management-api/tools/temporary-file/__tests__/helpers/example.jpg";
