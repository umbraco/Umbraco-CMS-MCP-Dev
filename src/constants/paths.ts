import { join } from 'path';

// Use absolute path from current working directory
export const RESOURCES_DIR = join(process.cwd(), 'build', 'resources').replace(/\\/g, '/');