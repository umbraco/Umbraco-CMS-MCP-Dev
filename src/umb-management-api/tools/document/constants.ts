export const UmbracoDocumentPermissions = {
    Create: "Umb.Document.Create",
    Read: "Umb.Document.Read",
    Update: "Umb.Document.Update",
    Delete: "Umb.Document.Delete",
    Publish: "Umb.Document.Publish",
    Unpublish: "Umb.Document.Unpublish",
    Move: "Umb.Document.Move",
    Sort: "Umb.Document.Sort",
    Duplicate: "Umb.Document.Duplicate",
    PublicAccess: "Umb.Document.PublicAccess",
    CultureAndHostnames: "Umb.Document.CultureAndHostnames",
    CreateBlueprint: "Umb.Document.CreateBlueprint",
}