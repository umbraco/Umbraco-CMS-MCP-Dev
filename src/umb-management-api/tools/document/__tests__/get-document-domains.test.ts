import GetDocumentDomainsTool from "../get/get-document-domains.js";
import { DocumentBuilder, TEST_DOMAIN } from "./helpers/document-builder.js";
import { DocumentTestHelper } from "./helpers/document-test-helper.js";
import { jest } from "@jest/globals";
import { BLANK_UUID } from "@/constants/constants.js";

const TEST_DOCUMENT_NAME = "_Test DomainsDocument";

describe("get-document-domains", () => {
  let originalConsoleError: typeof console.error;
  let docId: string;

  beforeEach(async () => {
    originalConsoleError = console.error;
    console.error = jest.fn();
    // Create a document and set domains
    const builder = await new DocumentBuilder()
      .withName(TEST_DOCUMENT_NAME)
      .withRootDocumentType()
      .create();
    await builder.setDomains([TEST_DOMAIN], null);
    docId = builder.getId();
  });

  afterEach(async () => {
    console.error = originalConsoleError;
    await DocumentTestHelper.cleanup(TEST_DOCUMENT_NAME);
  });

  it("should add and then get domains for a valid document", async () => {
    const result = await GetDocumentDomainsTool().handler(
      {
        id: docId,
      },
      { signal: new AbortController().signal }
    );
    expect(result).toMatchSnapshot();

    // Get the document back and check its domains
    const found = await DocumentTestHelper.findDocument(TEST_DOCUMENT_NAME);
    expect(found).toBeDefined();
    // Check if domain was added
    const domains = JSON.parse(result.content[0].text as string);
    expect(domains.domains).toEqual(
      expect.arrayContaining([expect.objectContaining(TEST_DOMAIN)])
    );
  });

  it("should handle non-existent document", async () => {
    const result = await GetDocumentDomainsTool().handler(
      {
        id: BLANK_UUID,
      },
      { signal: new AbortController().signal }
    );
    expect(result).toMatchSnapshot();
  });
});
