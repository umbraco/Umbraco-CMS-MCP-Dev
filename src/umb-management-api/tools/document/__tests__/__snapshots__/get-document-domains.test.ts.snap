// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-document-domains should add and then get domains for a valid document 1`] = `
{
  "content": [
    {
      "text": "{"defaultIsoCode":null,"domains":[{"domainName":"example.com","isoCode":"en-US"}]}",
      "type": "text",
    },
  ],
}
`;

exports[`get-document-domains should handle non-existent document 1`] = `
{
  "content": [
    {
      "text": "{"defaultIsoCode":null,"domains":[]}",
      "type": "text",
    },
  ],
}
`;
